test/hotspot/jtreg/compiler/c2/TestDivModNodes.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.ir_framework.Test;
+
+import java.util.function.BiFunction;
+
+/*
+ * @test
+ * @summary Test DIV and MOD nodes are converted into DIVMOD where possible
+ * @requires os.arch=="amd64" | os.arch=="x86_64"
```

Comment:
So this can run on aarch64 now?

Reply:
I was under the impression aarch64 cannot calculate quotient and remainder with a single div instruction (unlike x86), and I assumed that's why existing tests on divmod doesn't include this arch. Looking at the source, only `src/hotspot/cpu/x86/x86_64.ad` and `s390.ad` seems to contain divmod related instructions.

Reply:
I've added test configurations to reflect aarch64 with `+XX:+UseDivMod`